### PR TITLE
added rewards logic

### PR DESCRIPTION
**Pull request by adrianochuy**
image
image

### CODE REVIEW

File: app/controllers/completions_controller.rb [hunk 1]
```diff
@@ -5,9 +5,11 @@ class CompletionsController < ApplicationController
 
 	def create
 		@completion = @survey.completions.where(completer: current_user).first_or_create
+		@rewards_cash = @survey.rewards_cash.where(user_id: current_user.id, survey_id: @survey.id, rewards_cash_amount: @survey.cpi/2).first_or_create
```
Comment (adrianochuy) [inline app/controllers/completions_controller.rb:8]: CloutCash is hard coded to be cpi / 2, at the moment.

File: app/controllers/completions_controller.rb [hunk 2]
```diff
@@ -5,9 +5,11 @@ class CompletionsController < ApplicationController
...
+		@experience = @survey.experiences.where(user_id: current_user.id, survey_id: @survey.id, experience_amount: 10).first_or_create
```
Comment (adrianochuy) [inline app/controllers/completions_controller.rb:9]: CloutXP is 10 by defaullt

File: app/views/admin/users/index.html.haml [hunk 3]
```diff
@@ -23,11 +23,11 @@
...
-                %td= link_to user.first_name + " " + user.last_name, class:".whitespace-nowrap.px-3.py-4.text-sm.text-gray-500" 
-                %td= link_to user.email, class: ".whitespace-nowrap.px-3.py-4.text-sm.text-gray-500" 
-                %td= link_to user.organized_surveys.count, user_path(user), class:".whitespace-nowrap.px-3.py-4.text-sm.text-gray-500" 
-                %td.relative.whitespace-nowrap.py-4.pl-3.pr-4.text-right.text-sm.font-medium.sm:pr-6
+                %td.whitespace-nowrap.py-4.pl-4.pr-3.text-sm.font-medium.text-gray-900.sm= link_to user.id, admin_user_path(user)
+                %td.whitespace-nowrap.py-4.pl-4.pr-3.text-sm.font-medium.text-gray-900.sm= link_to user.full_name, admin_user_path(user)
+                %td.whitespace-nowrap.py-4.pl-4.pr-3.text-sm.font-medium.text-gray-900.sm= link_to user.email, admin_user_path(user)
+                %td.whitespace-nowrap.py-4.pl-4.pr-3.text-sm.font-medium.text-gray-900.sm= link_to user.organized_surveys.count, user_path(user)
+                %td.relative.whitespace-nowrap.py-4.pl-3.pr-4.text-right.text-sm.font-medium.sm
```
Comment (adrianochuy) [inline app/views/admin/users/index.html.haml:30]: fixed minor link error

File: app/views/profiles/_form.html.haml [hunk 4]
```diff
@@ -1,17 +1,11 @@
...
   .form-inputs
-    = f.input :age, collection: 0..99, prompt: "Select your age"
-    = f.input :gender, collection: Profile.select_options_for_gender, prompt: "Select your gender"
-    = f.input :ethnicity, collection: Profile.select_options_for_ethnicity, prompt: "Select your ethnicity"
-    = f.country_select :country
+    = f.input :age, collection: 0..99, prompt: "Select your age", error: 'Please select your age'
+    = f.input :gender, collection: Profile.select_options_for_gender, prompt: "Select your gender", error: 'Please select your gender'
+    = f.input :ethnicity, collection: Profile.select_options_for_ethnicity, prompt: "Select your ethnicity", error: 'Please select your ethnicity'
+    %p Country
+    = f.country_select :country, error: 'Please select a country'
```
Comment (adrianochuy) [inline app/views/profiles/_form.html.haml:9]: added error messages

File: db/migrate/20220901185127_add_fields_to_surveys.rb [hunk 5]
```diff
@@ -2,7 +2,7 @@ class AddFieldsToSurveys < ActiveRecord::Migration[7.0]
   def change
     add_column :surveys, :start_date, :datetime
     add_column :surveys, :end_date, :datetime
-    add_column :surveys, :cpi, :integer
+    add_column :surveys, :cpi, :decimal
```
Comment (adrianochuy) [inline db/migrate/20220901185127_add_fields_to_surveys.rb:5]: changed cpi to decimal, so it can handle currency

**Review by dowlingmi01 — APPROVED** [review]

**Pull request merged**